core/src/Cuda/Kokkos_Cuda_UniqueToken.cpp

```diff
@@ -0,0 +1,38 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#ifndef KOKKOS_IMPL_PUBLIC_INCLUDE
+#define KOKKOS_IMPL_PUBLIC_INCLUDE
+#endif
+
+#include <Kokkos_Core.hpp>
+#include <Cuda/Kokkos_Cuda_UniqueToken.hpp>
+
+namespace Kokkos::Experimental {
+
+UniqueToken<Cuda, UniqueTokenScope::Global>::UniqueToken(size_type max_size) {
+  m_locks = Kokkos::View<uint32_t*, Kokkos::CudaSpace>(
+      std::string("Kokkos::UniqueToken::m_locks"), max_size);
```

Comment:
Why did you add explicit std::string construction?

Reply:
Trying to have fewer template instantiations for different lengths of the C-style array. I'm happy to switch to C-style arrays again if you prefer.

Reply:
Please undo.  This denotes an issue with the View constructors, not something that should be dealt with at call site.

Reply:
f3b1667ce